src/PrestaShopBundle/Form/Admin/Sell/Address/CustomerAddressType.php

```diff
@@ -138,7 +139,7 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 'The national ID card number of this person, or a unique tax identification number.',
                 'Admin.Orderscustomers.Feature'
             ),
-            'required' => false,
+            'required' => in_array('dni', $requiredFields) ? true : false,
```

Comment:
```suggestion
            'required' => in_array('dni', $requiredFields),
```

should be enough because `in_array` returns boolean